metis-indexing/src/main/java/eu/europeana/indexing/mongo/property/MongoPropertyUpdaterImpl.java

```diff
@@ -273,22 +273,21 @@ public void updateWebResourceMetaInfo(Function<T, WebResourceMetaInfo> getter,
       updaterSupplier.get().delete(ancestorInformation, mongoServer);
     } else {
       updaterSupplier.get().update((WebResourceMetaInfoImpl) entity, ancestorInformation,
-              null, null, mongoServer);
+          null, null, mongoServer);
     }
   }
 
   @Override
   public T applyOperations() {
-    final UpdateOperator firstUpdateOperator = updateOperators.get(0);
-    final UpdateOperator[] extraUpdateOperators = this.updateOperators
-        .subList(1, this.updateOperators.size()).toArray(UpdateOperator[]::new);
-    final Update<T> update = queryCreator.get().update(firstUpdateOperator, extraUpdateOperators);
+
+    final UpdateOperator[] extraUpdateOperators = this.updateOperators.toArray(UpdateOperator[]::new);
     final UpdateOptions updateOptions = new UpdateOptions().upsert(true).multi(true);
+
     try {
-      retryableExternalRequestForNetworkExceptions(() -> update.execute(updateOptions));
+      retryableExternalRequestForNetworkExceptions(() -> queryCreator.get().update(updateOptions, extraUpdateOperators));
```

Comment:
I do not this the .get() should be in the try block, same for the catch block.
The try-catch here is meant for duplicate key exceptions and not other type of failures.

Reply:
oh I see. yes. modified to be outside the try-block.